gameserver/gameengine/export/game_engine.h

```diff
@@ -41,17 +43,20 @@
 
 class OutgoingPacket;
 class PlayerCtrl;
-class World;
 
 class GameEngine
 {
  public:
-  GameEngine(GameEngineQueue* gameEngineQueue, World* world, const std::string& loginMessage);
```

Comment:
Move constructor parameters to init(), then we can remove setGameEngine() in GameEngineQueue. Just create the GameEngine first, without parameters, then create GameEngineQueue with GameEngine* as parameter and finally call GameEngine::init() with GameEngineQueue*, loginMessage and everything that's already in init().